cds-plugin.js

```diff
@@ -89,7 +94,9 @@ cds.build?.register?.('typescript', class extends cds.build.Plugin {
 
     async #runCdsTyper () {
         DEBUG?.('running cds-typer')
-        await typer.compileFromFile('*', { outputDirectory: this.#modelDirectoryName })
+        cds.env.typer ??= {}
+        cds.env.typer.outputDirectory ??= this.#modelDirectoryName
```

Comment:
This does not have the expected effect, due to the fact that the module `./lib/compile` is implicitly loading module `./lib/config`.  So, the constructor of the `Config` class has already been executed and any changes to `cds.env.typer` will be ignored.

You would have to change it to

```js
const { configuration } = require('@cap-js/cds-typer/lib/config')
...
configuration.outputDirectory ??= this.#modelDirectoryName
```

Reply:
You're right, the properties should actually not be imprinted into the config at all, but dynamically queries from `cds.env`.
The joys of global state... I will check it again.

Reply:
(the fix should actually workd, as line 98 is basically a no-op, since `#modelDirectoryName` will attempt to read the `cds.typer.outputDirectory`, but it is still bad style)

Reply:
Ah, I see your use case now and I am afraid you are correct.
I have made the initialisation of `configuration` lazy to support modifications of `cds.env` before the first actual use of `cds-typer`.

Reply:
Looks good. The build runs through without any errors now.

Reply:
Great! Thanks for bringing this to our attention and for the review. 🙂